src/Components/test/Microsoft.AspNetCore.Components.Server.Test/Rendering/HtmlRendererTests.cs

```diff
@@ -0,0 +1,246 @@
+using System;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.Server.Rendering
+{
+    public class HtmlRendererTests
```

Comment:
Might be good to have a few more tests to cover scenarios like:

 * Region frames
 * ComponentRefCapture, ElementRefCapture (I know they are no-ops, but it's good to show that in tests)
 * HTML text encoding, HTML attribute encoding
 * Conditional attributes (e.g., `<button disabled=@someBoolFlag />`)

Reply:
@SteveSandersonMS Yeah, I'll add those.